src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/Constants.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.DotNet.Darc.Tests.VirtualMonoRepo;
+
+public class Constants
+{
+    public static readonly string VersionDetailsTemplate = 
+@"<?xml version=""1.0"" encoding=""utf-8""?>
+<Dependencies>
+    <ProductDependencies>
+        {0}
+    </ProductDependencies>
+    <ToolsetDependencies>
+    </ToolsetDependencies>
+</Dependencies>";
+
+    public static readonly string EmptyVersionDetails = 
+@"<?xml version=""1.0"" encoding=""utf-8""?>
+<Dependencies>
+    <ProductDependencies>
+    </ProductDependencies>
+    <ToolsetDependencies>
+    </ToolsetDependencies>
+</Dependencies>";
+    
+    public static readonly string DependencyTemplate = 
+@"<Dependency Name=""{0}"" Version=""8.0.0"">
+    <Uri>{1}</Uri>
+    <Sha>{2}</Sha>
+    <SourceBuild RepoName=""{0}"" ManagedOnly=""true"" />
+</Dependency>";
+
+    public static readonly string SourceMappingsTemplate =
+        @"{{
+  ""patchesPath"": ""{0}"",
+  ""additionalMappings"": [
+    {1}
+  ],
+
+  ""defaults"": {{
+      ""defaultRef"": ""main"",
+      ""exclude"": [
+        ""**/*.dll"",
+        ""**/*.Dll"",
+        ""**/*.exe"",
+        ""**/*.pdb"",
+        ""**/*.mdb"",
+        ""**/*.zip"",
+        ""**/*.nupkg""
+      ]
+    }},
+
+  ""mappings"": [
+    {2}
+  ]
+}}";
+
+    public static readonly string MappingTemplate =
```

Comment:
Did you consider having the `source-mappings.json` as C# objects and only serialize them into JSON? Might avoid this JSON kung-fu this way?

We even have classes for that:
https://github.com/dotnet/arcade-services/blob/4b97ba07fa0b7cbda645aa94f32099c4e3ef0710/src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/SourceMappingParser.cs#L155-L170

Reply:
they're private though. I can make them public and move them to the Models?

Reply:
Yes please